worker/caasbroker/manifold.go

```diff
@@ -55,10 +56,13 @@ func manifoldOutput(in worker.Worker, out interface{}) error {
 	if !ok {
 		return errors.Errorf("expected *broker.Tracker, got %T", in)
 	}
-	outBroker, ok := out.(*caas.Broker)
-	if !ok {
+	switch result := out.(type) {
+	case *caas.Broker:
+		*result = inTracker.Broker()
+	case *environs.CloudDestroyer:
+		*result = inTracker.Broker()
+	default:
 		return errors.Errorf("expected *caas.Broker, got %T", out)
```

Comment:
Maybe update the error as well?

Reply:
I thought about it and thought it could get too verbose. But maybe not. I'll update